src/mesh_connectivity.cpp

```diff
@@ -57,6 +57,92 @@ void compute_mesh_edges(const std::vector<PolygonFace>& mesh_faces,
 
 void compute_patches(const std::vector<std::vector<size_t>>& edges_of_face,
                      const std::vector<Edge>& mesh_edges,
+                     const std::vector<PolygonFace>& mesh_faces,
+                     const std::vector<IsoVert> iso_verts,
+                     std::vector<std::vector<size_t>>& patches,
+                     std::vector<size_t>& patch_function_label)
+{
+    std::vector<bool> visisted_face(edges_of_face.size(), false);
+    for (size_t i = 0; i < edges_of_face.size(); i++) {
+        if (!visisted_face[i]) {
+            // new patch
+            //std::cout << "new patch: " << std::endl;
+            patches.emplace_back();
+            auto& patch = patches.back();
+            std::queue<size_t> Q;
+            Q.push(i);
+            patch.emplace_back(i);
+            visisted_face[i] = true;
+            absl::flat_hash_map<size_t, size_t> func_to_num; //adding a hash map that finds the number of vertices that are associated to each function
```

Comment:
Is this information (number of vertices for each function within a patch) necessary? Is this this purely for a correctness check? If so, the logic should be moved to unit test.

Reply:
Thanks! I think we should try to keep the computation code separate from validation code. Especially when validation is not cheap.

If this logical cannot be moved to unit tests, we can refactor the check out into its own function, say `check_patch_label(...)` that return true only when the check passed. This way, we can `assert(check_patch_label(...))` in the main body of the code, and the check will only be executed in debug mode.